### PR TITLE
Update contact

### PR DESCRIPTION
**Pull request by adeelq**

### CODE REVIEW

File: Observer/Customer/CreateUpdateContact.php [hunk 1]
```diff
@@ -78,12 +79,26 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
+
             //email change detection
-            if ($email != $emailBefore) {
+            if ($emailBefore && $email != $emailBefore) {
+                //Reload contact model up update email
+                $contactModel = $this->contactFactory->create()
+                    ->loadByCustomerEmail($emailAddress, $websiteId);
+                $contactModel->setEmail($email);
+
                 $this->helper->log('email change detected : ' . $email
```
Comment (cdiacon) [inline Observer/Customer/CreateUpdateContact.php]: @adeelq Can change the log to : 

```
 $this->helper->log('email change detected from : ' . $emailBefore . ', to : ' . $email .
                    ', website id : ' . $websiteId
                );
```

**Comment by cdiacon** [comment]
@adeelq is this https://github.com/dotmailer/dotmailer-magento2-extension/pull/226/commits/5d0d82877628e204cd32e0e7efc49a396b002e67 

same as? : 

https://github.com/dotmailer/dotmailer-magento2-extension/pull/212/commits/0a1e2be72c67ff217aacd4061592d5a02637c7ff

**Comment by adeelq** [comment]
@cdiacon i did not look but yes it is indeed

**Comment by cdiacon** [comment]
@adeelq also need a check for suppressedContactIds not empty : 

https://github.com/dotmailer/dotmailer-magento2-extension/blob/f65375f8c35d790a8e0a80a22f4658dec1b2f0cf/Model/Sync/Contact/Update.php#L119-L119

**Pull request merged**